searchlib/src/vespa/searchlib/expression/and_predicate_node.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include "and_predicate_node.h"
+#include "resultnode.h"
+
+#include <algorithm>
+
+namespace search::expression {
+using vespalib::Deserializer;
+using vespalib::Serializer;
+
+IMPLEMENT_IDENTIFIABLE_NS2(search, expression, AndPredicateNode, MultiArgPredicateNode);
+
+AndPredicateNode::AndPredicateNode() noexcept = default;
+
+AndPredicateNode::~AndPredicateNode() = default;
+
+AndPredicateNode::AndPredicateNode(const AndPredicateNode&) = default;
+
+AndPredicateNode& AndPredicateNode::operator=(const AndPredicateNode&) = default;
+
+bool AndPredicateNode::allow(const DocId docId, const HitRank rank) {
+    return std::ranges::all_of(args(), [docId, rank](auto arg){ return arg->allow(docId, rank); });
```

Comment:
```suggestion
    return std::ranges::all_of(args(), [docId, rank](auto &arg){ return arg->allow(docId, rank); });
```
This (and the other 3 similar places) needs to operate on the wired argument, not on a copy of it.